### PR TITLE
MGMT-6484 [WIP] Create cache for post step reply api calls

### PR DESCRIPTION
**Pull request by ori-amizur**
In case a previous result exists in the cache and it is the same as the current result, do not send to the assisted service

/cc @tsorya

### CODE REVIEW

File: src/commands/reply_cache.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+var replyCache sync.Map
+
+func alreadyExistsInService(stepType models.StepType, value string) bool {
+	storedValue, ok := replyCache.Load(stepType)
+	return ok && funk.Equal(storedValue, value)
+}
+
+func storeInCache(stepType models.StepType, value string) {
```
Comment (tsorya) [inline src/commands/reply_cache.go:17]: don't we wanted to have cache with ttl?

File: src/commands/reply_cache.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package commands
+
+import (
+	"sync"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/thoas/go-funk"
+)
+
+var replyCache sync.Map
```
Comment (ronniel1) [inline src/commands/reply_cache.go:10]: I think we should use go-cache instead and have TTL on the entries

**Pull request merged**